src/expr/oracle_caller.h

```diff
@@ -43,37 +42,34 @@ class OracleCaller
    * fapp. Store in result res.
    *
    * Return true if the call was made, and false if it was already cached.
-   *
-   * If this method returns true, then runResult is set to the result returned
-   * from executing the binary.
    */
-  bool callOracle(const Node& fapp, Node& res, int& runResult);
-
-  /** Get the binary name for this oracle caller */
-  std::string getBinaryName() const;
+  bool callOracle(const Node& fapp, std::vector<Node>& res);
 
   /** Get cached results for this oracle caller */
-  const std::map<Node, Node>& getCachedResults() const;
-
-  /**
-   * Get binary from an oracle function, or an oracle interface quantified
-   * formula.
-   */
-  static std::string getBinaryNameFor(const Node& n);
+  const std::map<Node, std::vector<Node>>& getCachedResults() const;
 
   /** is f an oracle function? */
   static bool isOracleFunction(Node f);
   /** is n an oracle function application? */
   static bool isOracleFunctionApp(Node n);
 
+  /**
+   * Get oracle from an oracle function, or an oracle interface quantified
+   * formula. Returns a node of kind ORACLE if the associated oracle exists,
+   * or null otherwise.
```

Comment:
What do you mean?  There is a debug Assert?

Reply:
Yes. Is that assertion temporary?

Reply:
No, right now, you should not use this method on things that don't have an associated oracle, since this suggests that something went wrong, regardless this allows error handling at a higher level in production mode if needed.